### ISSUE TITLE
Selective Disclosure ZKPs for Supply Chain

### ISSUE THREAD

**Issue by OR13** [w3c-ccg/traceability-interop#20]
We added support for https://github.com/w3c-ccg/ldp-bbs2020

to the vc-http-api here: https://vc.transmute.world/api/docs

We should come up with some exemplar credentials that would be good to provide examples of using this proof suite.

An almost complete working PoC here: http://didme.me/

**Comment by OR13** [comment]
@mprorock can you add a flow chart / example covering multiple parties.

**Comment by OR13** [comment]
2 actions remain on this issue:

- example flow (real world use case)
- update vc-http-api tests to cover `deriveCredential`

**Comment by OR13** [comment]
Proposal: we should make `deriveCredential` a MUST for trace-interop

**Comment by TallTed** [comment]
This was the nearest issue to the discussion in IRC, but I'm not sure it's actually what we were talking about, where I said `mock PROPOSAL: trace-interop implementations MUST negotiate encryption mechanisms by requesting ____ and replying ____ ` 

Candidate response header, available in HTTP/1.x and HTTP/2; will presumably persist to HTTP/3 --
>**`Pragma`**
`Pragma: no-cache`
Implementation-specific fields that may have various effects anywhere along the request-response chain.

Could also dive much deeper and go with [`5.5.  Extending HTTP/2`](https://datatracker.ietf.org/doc/html/rfc7540#section-5.5).

Neither of these is quite what I was thinking of, so I'll be digging a bit more, too.

**Comment by TallTed** [comment]
Issuing `HEAD` against specific URIs can get returned `Link:` headers which have complex values, such as --

`Link: <{ URI of encryption endpoint }>; rel="{ plain-literal as registered with IANA (registration recommended but not absolutely required), or URI as defined by this spec; e.g., "BBS+" or "https://github.com/w3c-ccg/ldp-bbs2020" }"; anchor="#if_needed"`

The `rel` value may be open-ended, allowing for extension as required by deployments, without requiring that they go through a registry -- allowing high-speed evolution when necessary, as well as private extensions that may not be understood by those not directly involved.

**Comment by OR13** [comment]
On the call today, we discussed the need to support discovery of both "issuer" ids and "credential suites".

We discussed the proposal to use the Link header, and `.well-known` resources.

**Comment by OR13** [comment]
We agreed, to open an issue on the vc-http-api to tackle these issues.

**Comment by mprorock** [comment]
big fan of http `accept` headers for http calls, as well as use of .well-known for capabilities discovery

**Comment by brianorwhatever** [comment]
We could also include this within a did:web document on the issuer `.well-known/did.json` endpoint. For example this did could include a service block that contains the endpoint and what crypto suites it uses and issuer ids managed at that endpoint

**Comment by OR13** [comment]
I opened this issue: https://github.com/w3c-ccg/vc-http-api/issues/218

**Comment by mkhraisha** [comment]
Discussed on call waiting on vc-http-api group action

**Comment by nissimsan** [comment]
Keep open, blocked by VC-API

**Comment by OR13** [comment]
There are known issues with BBS+, we should close this issue until its possible to use any LD Proof format that supports derivation.

See https://github.com/w3c-ccg/ldp-bbs2020/issues/62

**Comment by brianorwhatever** [comment]
This issue is closed because the upstream problem needs to be resolved. We will open a new issue once that has been resolved.

**Comment by TallTed** [comment]
My concern is "What triggers a new issue creation, upon resolution of the blocker?" which flows to "Who or what is watching for resolution of that blocker?"

**Comment by brianorwhatever** [comment]
@TallTed I definitely see where you're coming from and have opened a new issue. I will continue to monitor this issue
